src/test/mock/clitools.rs

```diff
@@ -703,54 +694,6 @@ impl Config {
         output
     }
 
-    #[cfg_attr(feature = "otel", tracing::instrument(skip_all))]
-    pub(crate) fn run_inprocess<I, A>(&self, name: &str, args: I, env: &[(&str, &str)]) -> Output
```

Comment:
Why is this necessary, and what is the effect on test suite execution time?

Reply:
@djc This has been explained in the final part of the post.

Reply:
@djc I mean if we can get rid of `static PROCESS` them there should be a way to revive #2367.

Reply:
Okay, I've been working on what I think might be an alternative path today. Let me push that up in a bit so we can decide which way to go.

Reply:
@djc The thing is that I'm not quite sure about what goal #2367 was trying to reach. Is it for perf optimization? If it is, then the positive impact does not seem that large, at least so far, and it seems to me that it's adding complexity with the dance altering `static`s. Is it for something else? The more I change this part without knowing the original intent the more afraid I am of accidentally breaking something in the process.

Reply:
My alternative is in #3871.

Reply:
No worries!